code/modules/mafia/controller.dm

```diff
@@ -542,16 +542,13 @@
 /datum/mafia_controller/proc/create_bodies()
 	for(var/datum/mafia_role/role in all_roles)
 		var/mob/living/carbon/human/H = new(get_turf(role.assigned_landmark))
+		ADD_TRAIT(H, TRAIT_NO_SELFCOMBUSTION, "mafia")
```

Comment:
I'm not sure if this works and I couldn't test it easily, so if anyone can give feedback it will be appreciated.

Reply:
It looks good. You should define the "mafia" string in `code/__DEFINES/traits.dm` under something like `#define MAFIA_TRAIT "mafia"`, and then change this to
```suggestion
		ADD_TRAIT(H, TRAIT_NO_SELFCOMBUSTION, MAFIA_TRAIT)
```